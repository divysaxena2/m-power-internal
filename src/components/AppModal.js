import React, { Component } from 'react';
import { Text, View, Button, Modal, StyleSheet, Image } from 'react-native';
import { Left, Right, Body, ListItem, List } from 'native-base'

export default class AppModal extends Component {
    state = {
        modalVisible: false,
    };

    openModal() {
        this.setState({ modalVisible: true });
    }

    closeModal() {
        this.setState({ modalVisible: false });
    }

    render() {
        return (
            <View style={styles.container}>
                <Modal
                    visible={this.state.modalVisible}
                    animationType={'slide'}
                    onRequestClose={() => this.closeModal()}>
                    <View style={styles.modalContainer}>
                        <View style={styles.innerContainer}>
                            <Image source={require('../assets/paymentSuccess.png')} />
                            <Text>Payment Successful!</Text>
                            <Text>Your Bill Details</Text>
                            <List>
                                <ListItem>
                                    <Text style={{ textAlign: 'right' }}>Provider Name</Text>
                                    <Text style={{ paddingLeft: 29 }}>El Camino Hospital</Text>
                                </ListItem>
                                <ListItem>
                                    <Text style={{ textAlign: 'left' }}>Amount</Text>
                                    <Text style={{ paddingLeft: 29 }}>$500</Text>
                                </ListItem>
                                <ListItem>
                                    <Text style={{ textAlign: 'right' }}>Payment Method</Text>
                                    <Text style={{ paddingLeft: 29 }}>Credit card ending 3924</Text>
                                </ListItem>
                            </List>
                            <View>
                                <Text>Congrats!You have earned 445 MPowered Points!</Text>
                            </View>
                            <Button
                                onPress={() => this.closeModal()}
                                title="Go to Bill Summary">
                            </Button>
                        </View>
                    </View>
                </Modal>
                <Button
                    onPress={() => this.openModal()}
                    title="Open modal"
                />
            </View >
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
    },
    modalContainer: {
        flex: 1,
        justifyContent: 'center',
        backgroundColor: 'white',
    },
    innerContainer: {
        alignItems: 'center',
    },
});