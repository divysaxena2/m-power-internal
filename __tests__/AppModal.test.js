import 'react-native';
import React from 'react';
import AppModal from '../src/components/AppModal';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer.create(
        <AppModal />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});