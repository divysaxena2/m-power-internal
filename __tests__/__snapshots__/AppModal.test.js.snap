// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    visible={false}
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <Image
          source={1}
        />
        <Text>
          Payment Successful!
        </Text>
        <Text>
          Your Bill Details
        </Text>
        <View
          closeOnRowBeginSwipe={false}
          closeOnRowPress={true}
          closeOnScroll={true}
          directionalDistanceChangeThreshold={2}
          disableLeftSwipe={false}
          disableRightSwipe={false}
          leftOpenValue={0}
          previewFirstRow={false}
          recalculateHiddenLayout={false}
          rightOpenValue={0}
          style={Object {}}
          swipeToOpenPercent={50}
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#c9c9c9",
                  "flexDirection": "row",
                  "marginLeft": 16,
                  "paddingBottom": 13,
                  "paddingRight": 16,
                  "paddingTop": 13,
                }
              }
            >
              <Text
                style={
                  Object {
                    "textAlign": "right",
                  }
                }
              >
                Provider Name
              </Text>
              <Text
                style={
                  Object {
                    "paddingLeft": 29,
                  }
                }
              >
                El Camino Hospital
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#c9c9c9",
                  "flexDirection": "row",
                  "marginLeft": 16,
                  "paddingBottom": 13,
                  "paddingRight": 16,
                  "paddingTop": 13,
                }
              }
            >
              <Text
                style={
                  Object {
                    "textAlign": "left",
                  }
                }
              >
                Amount
              </Text>
              <Text
                style={
                  Object {
                    "paddingLeft": 29,
                  }
                }
              >
                $500
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderBottomWidth": 0.5,
                  "borderColor": "#c9c9c9",
                  "flexDirection": "row",
                  "marginLeft": 16,
                  "paddingBottom": 13,
                  "paddingRight": 16,
                  "paddingTop": 13,
                }
              }
            >
              <Text
                style={
                  Object {
                    "textAlign": "right",
                  }
                }
              >
                Payment Method
              </Text>
              <Text
                style={
                  Object {
                    "paddingLeft": 29,
                  }
                }
              >
                Credit card ending 3924
              </Text>
            </View>
          </View>
        </View>
        <View>
          <Text>
            Congrats!You have earned 445 MPowered Points!
          </Text>
        </View>
        <View
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Go to Bill Summary
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Open modal
      </Text>
    </View>
  </View>
</View>
`;
